service/Microsoft.Azure.Devices/ServiceClient.cs

```diff
@@ -44,6 +45,26 @@ public static ServiceClient CreateFromConnectionString(string connectionString)
             return CreateFromConnectionString(connectionString, TransportType.Amqp);
         }
 
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases unmanaged and - optionally - managed resources.
+        /// </summary>
+        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if ( disposing )
+            {
+                // Call CloseAsync() synchronously avoiding AggregateException.
+                this.CloseAsync().GetAwaiter().GetResult();
+            }
+        }
```

Comment:
You are right. But adding abstract methods to public class causes breaking change (It may hurt people who already have derived classes such as hand made mock objects.)
Would you tell me why you think the base impls should be removed?